### ISSUE TITLE
time_stretch aiming at sample sizes < 1000

### ISSUE THREAD

**Issue by jwb95** [librosa/librosa#900]
Hi,

I wish to time-stretch arrays with lengths between 150 and 1000 samples to new arrays with lengths within the same range.

However, e.g. when using `librosa.effects.time_stretch` on an array with `length=300` I would supply `rate=0.5` in order to get an output with `length=600`.
Unfortunately, in this case the length of the output would be =512. In other cases I would end up with a length of 0.

Is it possible to overcome this?

Thank you

**Comment by lostanlen** [comment]
The reason why this happens is that `librosa.effects.time_stretch` calls `librosa.stft` with default parameters, and 512 is the default `hop_length`.

Right now the quickest way to fix this is to compute this STFT yourself with non-default parameters, and then apply the phase vocoder in the custom STFT domain.

```
stft = librosa.core.stft(y, n_fft=128)
stft_stretch = librosa.core.phase_vocoder(stft, rate)
y_stretch = librosa.core.istft(stft_stretch, dtype=y.dtype)
```

Note that the ratio `len(y)/len(y_stretch)` will be close to `rate`, but not exactly equal. You might have to do some trimming post hoc. Maybe that is something that we can improve in the wake of #854 ?

I'm curious why you'd want to apply a phase vocoder to such a small array though. Is it a very short audio frame?

**Comment by lostanlen** [comment]
@jwb95 also note that pyrubberband may be more appropriate for your needs:
https://github.com/bmcfee/pyrubberband

**Comment by bmcfee** [comment]
> pyrubberband may be more appropriate for your needs

Can confirm: librosa's built-in vocoder is extremely basic, and is primarily intended as a reference implementation for educational purposes.  Rubberband will give you much better quality.

**Comment by lostanlen** [comment]
closed by #901
